src/libraries/Common/src/System/Net/LazyAsyncResult.cs

```diff
@@ -253,7 +251,7 @@ internal bool InternalPeekCompleted
         }
 
         // Internal property for setting the IO result.
-        internal object Result
+        internal object? Result
```

Comment:
``` csharp
    if (value == DBNull.Value)
    {
        NetEventSource.Fail(this, "Result can't be set to DBNull - it's a special internal value.");
    }
```
Makes me think its not allow null, no?

Reply:
DBNull is different from null.

Reply:
(That said, I wouldn't spend too much thought on this file... I'm hopeful we can delete it outright in the not too distant future.)